test/http2/secure-with-fallback.test.js

```diff
@@ -9,97 +9,100 @@ const h2url = require('h2url')
 const sget = require('simple-get').concat
 const msg = { hello: 'world' }
 
-let fastify
-try {
-  fastify = Fastify({
-    http2: true,
-    https: {
-      allowHTTP1: true,
-      key: fs.readFileSync(path.join(__dirname, '..', 'https', 'fastify.key')),
-      cert: fs.readFileSync(path.join(__dirname, '..', 'https', 'fastify.cert'))
-    }
-  })
-  t.pass('Key/cert successfully loaded')
-} catch (e) {
-  t.fail('Key/cert loading failed', e)
-}
-
-fastify.get('/', function (req, reply) {
-  reply.code(200).send(msg)
-})
-
-fastify.post('/', function (req, reply) {
-  reply.code(200).send(req.body)
-})
+const { buildCertificate } = require('../build-certificate')
+buildCertificate().then(function () {
```

Comment:
This is actually very ugly in my mind.
As the `instance` is started and modified outside the `t.test`, I need to wrap it like this to ensure the `buildCertificate` is actually waited before test is started.

Anyone have a better implementation is appreciate.

Reply:
The certificate generation should be performed something like:

```js
let generatedCerts = false
tap.beforeEach(async t => {
  if (generatedCerts === false) {
    await generateCerts()
    generatedCerts = true
  }

  t.context.cert = cert
  t.context.key = key
})
```

Reply:
Have you deeply look at the test file? How can `t.before` can be ensure to run before the assertion without `t.test` wrapping for the test case?

Reply:
I haven't looked at the test file at all, really. It might need to be refactored and that would be okay.

Reply:
I am going to wrap it with a big `test` and I can use `t.before` in this case.